ark-testing/src/daemon/mod.rs

```diff
@@ -129,13 +131,28 @@ where T : DaemonHelper + Send + Sync + 'static
 			}
 		}));
 
-		self.child = Some(child);
-
-		// TODO: Provide a time-out here
-		// TODO: Check if the child has finished after every await
-		self.inner.wait_for_init().await?;
+		// Wait for init
+		// But check every 100 milliseconds if the Child is
+		// still running
+		let duration = std::time::Duration::from_millis(100);
+		let mut is_initialized = false;
+		while is_running(&mut child) && ! is_initialized {
+			match tokio::time::timeout(duration, self.inner.wait_for_init()).await {
+				Err(_) => {},
+				Ok(result) => {
+					result?;
+					is_initialized = true;
+				}
+			}
+		}
 
-		Ok(())
+		if is_initialized {
+			self.child = Some(child);
+			Ok(())
+		}
+		else {
+			anyhow::bail!("Failed to initialize {}", self.inner.name());
```

Comment:
Can we do `#[macro_use] extern crate anyhow;` wherever we need so we can use `bail!()` directly? Or do you prefer this to be explicit? I don't mind either, but would like to be somewhat consistent.

I know we use `anyhow::Result`, but (a) it conflicts with stdlib Result and (b) it's not a macro so we'd have to have use statements in each file for it. The macro use is only at the crate root.

Reply:
Both options are fine for me.
Do you want me to make the change?

Reply:
Yeah, we use the macro directly everywhere else. Thanks.